java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java

```diff
@@ -726,6 +724,26 @@ private K disjunctResults(List<K> results) {
                         nextQueue.add(result);
                     }
                 }
+
+                if (Thread.currentThread().isInterrupted()) {
+                    throw KEMException.criticalError("Thread interrupted");
+                }
+                    // DISABLE EXCEPTION CHECKSTYLE
+                } catch (OutOfMemoryError e) {
```

Comment:
can we just catch all Error and immediately rethrow them? there are probably a bunch of other Errors that we probably don't want to try to execute code after they're thrown. And there's no way that an allocation is going to work after you catch an OutOfMemoryError.

Reply:
Then what's the point of catching them? There are many errors of interest: AssertionError and StackOverflowError in particular. They can't cause issues here.
The only error that could be hidden at this point by a new error is OutOfMemoryError. There is still a big chance that printSummaryBox here will work, and I've seen it working on Jenkins. A little allocation is possible because some pointers get released from the moment exception is thrown till the moment it is caught. And most OOME is GC limit exceeded - that means there is still a bit of free memory left but GC takes almost 100% of the time.

And I do more clever things with OOME in another PR that implements advanced memory profiling.

Reply:
You name AssertionError and StackOverflowError in particular. I agree that these can probably be caught, but why not catch them explicitly rather than catching all errors?

Reply:
I named these 2 because I encountered them explicitly. What's wrong with catching Throwable? If we ever encounter problems because some other error is caught, this clause can be refined. But for now, maybe it's more useful this way.

Reply:
This looks like a typical case of law of triviality.

Reply:
Ok I changed as you asked. Catching RuntimeException, AssertionError, StackOverflowError.